src/playground/sample_data_ingestion/jobs/startup_job.py

```diff
@@ -0,0 +1,145 @@
+                timestamp = None
+            else:
+                timestamp = config["timestamp"]
+            index_body = config["index_body"]
+            feature = config["feature"]
+            anomaly_percentage = float(config["anomaly_percentage"])
+            avg_min = config["avg_min"]
+            avg_max = config["avg_max"]
+            minutes = int(config["minutes"])
+            abs_min = config["abs_min"]
+            abs_max = config["abs_max"]
+            days_before = int(config["days_before"])
+            days_after = int(config["days_after"])
+            calculated_date = date.today()
+            create_payload = config["create_payload"]
+            file_provided = config["file_provided"]
+            plugin = config["plugin"]
+
+            # Generate date range of indices (or just 1 if days_after and days_before is 0)
+            for i in range(days_after + days_before + 1):
+                current_date = date.today()
+                can_ingest = True
+                index_name_to_create = index_name
+
+                # If user specifies minute intervals, number variable will be updated
+                if minutes:
+                    number = ceil(MINUTES_PER_DAY/minutes)
+
+                # If user specifies a date range, index names with dates are appended
+                if days_after or days_before:
+                    calculated_date = date.today() - timedelta(days = (days_before - i))
+                    calculated_date = datetime(calculated_date.year, calculated_date.month, calculated_date.day)
+                    index_name_to_create = index_name + "_" + str(calculated_date.month) + "_" + str(calculated_date.day) + "_" + str(calculated_date.year)
+                    current_date = calculated_date
+
+                # Whether the index exists or not, a new index object is created
+                new_index = SampleDataIndex(index_name_to_create, client, index_body)
+
+                # If an index exists, we do not ingest more data as it is assumed that the data is already ingested
+                if not client.indices.exists(index = index_name_to_create):
+                    new_index.create_index()
+                    new_index.ingest_more(
+                      file_provided = file_provided, 
+                      template = template, 
+                      mapping = mapping, 
+                      index_name = index_name_to_create, 
+                      number = number, 
+                      chunk = chunk, 
+                      timestamp = timestamp, 
+                      feature = feature, 
+                      anomaly_percentage = anomaly_percentage,
+                      avg_min = avg_min, 
+                      avg_max = avg_max, 
+                      abs_min = abs_min, 
+                      abs_max = abs_max, 
+                      minutes = minutes,
+                      current_date = current_date
+                    )
+        URL = "https://" + HOST
```

Comment:
Is it possible to use urljoin? Thanks

Reply:
Yes, and I have modified `URL` to accept a `scheme` and `host`, which both can be configured as CLI arguments.